core/src/main/java/org/apache/iceberg/encryption/BaseEncryptedKey.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.encryption;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+public class BaseEncryptedKey implements EncryptedKey {
+  private final String keyId;
+  private final ByteBuffer keyMetadata;
+  private final String encryptedById;
+  private final Map<String, String> properties;
+
+  public BaseEncryptedKey(
+      String keyId, ByteBuffer keyMetadata, String encryptedById, Map<String, String> properties) {
+    this.keyId = keyId;
+    this.keyMetadata = keyMetadata;
```

Comment:
since keyId + metadata are required by the spec, should this be verifying that they aren't null?